File: exp-01/q2-max-min-recursion/solution.js
const readline = require('readline');

const rl = readline.createInterface({ input: process.stdin, output: process.stdout });

function findMinMax(arr, n) {
    if (n === 1) return [arr[0], arr[0]];
    const [curMin, curMax] = findMinMax(arr, n - 1);
    return [Math.min(curMin, arr[n - 1]), Math.max(curMax, arr[n - 1])];
}

const lines = [];
rl.on('line', (line) => {
    lines.push(line.trim());
    if (lines.length === 2) {
        const n = parseInt(lines[0], 10);
        const arr = lines[1].split(/\s+/).map(Number);
        const [minVal, maxVal] = findMinMax(arr, n);
        console.log(`Min: ${minVal}`);
        console.log(`Max: ${maxVal}`);
        rl.close();
    }
});

File: exp-01/q2-max-min-recursion/solution.py
def find_min_max(arr, n):
    if n == 1:
        return arr[0], arr[0]
    cur_min, cur_max = find_min_max(arr, n - 1)
    cur_min = min(cur_min, arr[n - 1])
    cur_max = max(cur_max, arr[n - 1])
    return cur_min, cur_max


n = int(input())
arr = list(map(int, input().split()))
min_val, max_val = find_min_max(arr, n)
print(f"Min: {min_val}")
print(f"Max: {max_val}")